platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -238,6 +238,31 @@ typedef void (^MGLOfflinePackRemovalCompletionHandler)(NSError * _Nullable error
  */
 @property (nonatomic, readonly) unsigned long long countOfBytesCompleted;
 
+/**
+ Sets a URL template for the given scheme.
+
+ Any URLs encountered in styles and subsequently loaded sources will be fed through
+ a URL replacement mechanism that e.g. transforms `mapbox://` URLs into regular
+ `https://` URLs. You can provide URL replacement patterns for your own scheme,
+ e.g. `example://`.
+
+ Any query string will be preserved from the original URL and merged with a
+ query string of the template URL, if any.
+
+ @param tpl A templated URL. it may contain these brace-enclosed replacement
+    tokens, e.g. for `"http://example.com/foo/bar/baz.png"`, valid tokens are:
+    * `{scheme}`: `"http"`
+    * `{domain}`: `"example.com"`
+    * `{path}`: `"foo/bar/baz.png"` (without leading `/`)
+    * `{directory}`: `"foo/bar/"` (with trailing `/`)
+    * `{filename}`: `"baz"` (without file extension)
+    * `{extension}`: `".png"` (optionally with a preceding `@2x` ratio specifier)
```

Comment:
Platform deviation would certainly be a problem when it comes to parsing URLs in JSON files, but a block would consist of platform-specific (or at least language-specific) code. Any deviation from how URLs are handled on the platform or in the language’s standard library would be more of a surprise, wouldn’t it?

In the unlikely event that incompatibilities between standard URL parsers on different platforms poses a problem to the developer, they could do minimal parsing inline. For most scenarios, however, a callback would enable developers to bring more context into the transformation logic without requiring us to add a special case for every innovation in tile URL technology.

Even if you don’t want to leave URL parsing up to the developer, why not have the block accept a dictionary mapping e.g. `directory` to `foo/bar/` and allow the developer to perform concatenation and any other logic themselves?